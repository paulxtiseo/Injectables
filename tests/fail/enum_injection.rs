use injectables::{injectable, inject_fields};

#[injectable]
pub enum Invalid {
    Variant1,
    Variant2,
}

#[inject_fields(Invalid)]
pub struct Test {
    pub name: String,
}

fn main() {}