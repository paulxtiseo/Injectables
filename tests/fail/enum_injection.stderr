error: Only structs are supported
 --> tests/fail/enum_injection.rs:3:1
  |
3 | #[injectable]
  | ^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `injectable` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot inject fields from 'Invalid' as this type does not exist
 --> tests/fail/enum_injection.rs:9:1
  |
9 | #[inject_fields(Invalid)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
